test/Playwright/pageObjects/BasePage.ts

```diff
@@ -0,0 +1,36 @@
+import { type Locator, type Page } from "@playwright/test";
+
+export class BasePage {
+  readonly page: Page;
+
+  constructor(page: Page) {
+    this.page = page;
+  }
+
+  async goto(url: string) {
+    await this.page.goto(url);
+    await this.fullyLoadPage();
+  }
+  async fullyLoadPage() {
+    await this.page.evaluate(() =>
+      document.querySelectorAll("img[loading=lazy]").forEach((img) => img.setAttribute("loading", "eager"))
+    );
+
+    await this.page.evaluate(async () => {
+      await new Promise((resolve) => {
+        var totalHeight = 0;
```

Comment:
We should use `const` here instead of `var`.

Reply:
On the line 26 there is:
```
totalHeight += distance;
```
So it should not be const, should it?

Reply:
Sure, sorry I missed this one. However, `distance` and `timer`, as well as `scrollHeight` should be declared as `const`. I would also prefer to declare `totalHeight` as `let` not `var` since `var` has function-scope and not block-scope which might confuse people. Also it's generally recommended to use `let` instead of `var` in cases when we need to reassign the value of variable in modern javascript.

I would suggest to do something like this here:

`import { type Locator, type Page } from "@playwright/test";

export class BasePage {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async goto(url: string) {
    await this.page.goto(url);
    await this.fullyLoadPage();
  }

  async fullyLoadPage() {
    await this.page.evaluate(() =>
      document.querySelectorAll("img[loading=lazy]").forEach((img) => img.setAttribute("loading", "eager"))
    );

    await this.page.evaluate(async () => {
      await new Promise((resolve) => {
        let totalHeight = 0;
        const distance = 200;
        const timer = setInterval(() => {
          const scrollHeight = document.body.scrollHeight;
          window.scrollBy(0, distance);
          totalHeight += distance;
          if (totalHeight >= scrollHeight - window.innerHeight) {
            clearInterval(timer);
            resolve(void 0);
          }
        }, 70);
      });
      window.scrollTo(0, 0);
    });
  }
}`

Reply:
Upraveno.